targets/TARGET_WICED/WICED.md

```diff
@@ -0,0 +1,229 @@
+   - `targets/TARGET_WICED/WICED/network/NuttX_NS`
+
+1. Add the `w_` prefix to all c/h files in WICED.
+
+   Edit all c/h files and change the relevant `#include` statments to use the
+   updated file names.
+
+   This is required to avoid naming conflicts between Mbed OS and the WICED SDK.
+
+1. Prevent Mbed OS from compiling the include-only c files. These are
+   any include statments that refer to a file ending in `.c`.
+
+   For example:
+   
+   ``` diff
+   - libraries/crypto/micro-ecc/w_asm_arm.c
+   + libraries/crypto/micro-ecc/w_asm_arm.inc
+   ```
+
+   Edit all c/h files, and change the relevant `#include` statements to use the
+   updated file names.
+
+1. Remove any references to the WICED file system.
+
+   For example:
+   
+   ``` diff
+   - #include "wicedfs.h"
+   ```
+
+1. Create a C wrapper for the firmware image.
+
+   In `targets/TARGET_WICED/WICED/resources/firmware/4343W`:
+   
+   ``` bash
+   xxd -i 4343WA1.bin > 4343WA1.c
+   sed -i 's/__4343WA1_bin/wifi_firmware_image_data/' 4343WA1.c
+   sed -i 's/unsigned/const unsigned/' 4343WA1.c
+   ```
+
+   Then, modify the firmware image in the C wrapper to match WICED's link-time
+   resource API.
+
+   In `targets/TARGET_WICED/WICED/resources/firmware/4343W/4343WA1.c`:
+   
+   ``` c
+   #include "w_resources.h"
+
+   extern const resource_hnd_t wifi_firmware_image;
+   extern const char wifi_firmware_image_data[383156];
+
+   const resource_hnd_t wifi_firmware_image = {
+       .location = RESOURCE_IN_MEMORY,
+       .size = sizeof(wifi_firmware_image_data),
+       .val.mem.data = wifi_firmware_image_data,
+   };
+
+   const char wifi_firmware_image_data[383156] = {
+       ...
+   ```
+
+   Then, create a `w_resources.h` file to reference the above firmware image.
+
+   ``` c
+   #ifndef INCLUDED_RESOURCES_H_
+   #define INCLUDED_RESOURCES_H_
+   #include "wiced_resource.h"
+   extern const resource_hnd_t wifi_firmware_image;
+   extern const char wifi_firmware_image_data[383156];
+   #endif
+   ```
+
+1. Create the `w_generated_security_dct.h` file that
+   the WICED SDK tools would normally generate.
+
+   ``` c
+   #define CERTIFICATE_STRING "\0"
+   #define PRIVATE_KEY_STRING "\0"
+   ```
+
+1. Create the `w_generated_mac_address.txt` file that
+   the WICED SDK tools would normally generate.
+
+   ``` c
+   #define NVRAM_GENERATED_MAC_ADDRESS        "macaddr=02:0A:F7:11:b6:19"
+   #define DCT_GENERATED_MAC_ADDRESS          "\x02\x0A\xF7\x19\x11\xb6"
+   #define DCT_GENERATED_ETHERNET_MAC_ADDRESS "\x02\x0A\xF7\x19\x11\xb7"
+   ```
+
+1. Change any instances of `_irq` functions to match Mbed OS's linker `_IRQHandler`
+
+   For example:
+   
+   ``` diff
+   - WWD_RTOS_DEFINE_ISR( NoOS_systick_irq )
+   + WWD_RTOS_DEFINE_ISR( NoOS_systick_IRQHandler )
+   ```
+
+1. Comment out the bad `wwd_bus_packet_available_to_read` call while polling for
+   bus packet status.
+
+   In `targets/TARGET_WICED/WICED/WWD/internal/wwd_thread.c` line 336:
```

Comment:
line 344

Reply:
@SeppoTakalo pointed out we may be on different versions. It should have been marked as 4.1.*. With 4.1 are these lines still different?